.github/workflows/check-pr.yml

```diff
@@ -0,0 +1,96 @@
+name: check-pr
+
+on:
+  pull_request:
+
+jobs:
+  build-and-test:
+    runs-on: [ ubuntu-22.04 ]
+    steps:
+      - name: checkout v4
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```

Comment:
any reason to stick to particular git sha? why not just `actions/checkout@v4`?

Reply:
It's a best practice to use specific sha ids for actions so we don't come up with unexpected breaking changes if they're introduced by actions authors.

That is to say, we need to bump this to the current version and test.

Reply:
uhm, any references for this? I'm seeing this advice for 3rd party actions only:

https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

Reply:
(we use standard actions a lot, and github tends to warn you if you're using something old that is about to turn obsolete - right at workflow job results page)

Reply:
I don't see why we should apply a different strategy for second-party (GH) actions as opposed to third-party (anyone else).  It's still not something we directly control.

Reply:
okay, fair enough (but still weird)